website/src/components/showcase/ShowcaseTagSelect/index.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {
+  ComponentProps,
+  ReactNode,
+  ReactElement,
+  useCallback,
+  useRef,
+  useEffect,
+} from 'react';
+import {useHistory, useLocation} from '@docusaurus/router';
+import {toggleListItem} from '@site/src/utils/jsUtils';
+import type {TagType} from '@site/src/data/users';
+
+import styles from './styles.module.css';
+
+interface Props extends ComponentProps<'input'> {
+  icon: ReactElement<ComponentProps<'svg'>>;
+  label: ReactNode;
+  tag: TagType;
+}
+
+const TagQueryStringKey = 'tags';
+
+export function readSearchTags(search: string): TagType[] {
+  return new URLSearchParams(search).getAll(TagQueryStringKey) as TagType[];
+}
+
+function replaceSearchTags(search: string, newTags: TagType[]) {
+  const searchParams = new URLSearchParams(search);
+  searchParams.delete(TagQueryStringKey);
+  newTags.forEach((tag) => searchParams.append(TagQueryStringKey, tag));
+  return searchParams.toString();
+}
+
+const ShowcaseTagSelect = React.forwardRef<HTMLLabelElement, Props>(
+  ({id, icon, label, tag, ...rest}, ref) => {
+    const location = useLocation();
+    const history = useHistory();
+    const inputRef = useRef<HTMLInputElement>();
```

Comment:
same, I don't understand why a ref is needed here

Reply:
Same, maybe there is a reason @Josh-Cena used ref. Else I'd update the component to use toggle state.